# HG changeset patch
# User Rik <rik@octave.org>
# Date 1452883968 28800
#      Fri Jan 15 10:52:48 2016 -0800
# Node ID 40f931a63a917ae42affb40fad6be4c9ece1a39a
# Parent  b433f99904527ea66c44d3a34243dcf708af63e8
Undo incorrect if/else inversion in cset 3aa293be0e8d.

* debug.cc (do_dbtype): Undo incorrect if/else inversion since the code does
not use error.

diff --git a/libinterp/corefcn/debug.cc b/libinterp/corefcn/debug.cc
--- a/libinterp/corefcn/debug.cc
+++ b/libinterp/corefcn/debug.cc
@@ -842,34 +842,38 @@ is stopped.\n\
   return ovl ();
 }
 
 void
 do_dbtype (std::ostream& os, const std::string& name, int start, int end)
 {
   std::string ff = fcn_file_in_path (name);
 
-  if (! ff.empty ())
+  if (ff.empty ())
     os << "dbtype: unknown function " << name << "\n";
-
-  std::ifstream fs (ff.c_str (), std::ios::in);
-
-  if (! fs)
-    os << "dbtype: unable to open '" << ff << "' for reading!\n";
+  else
+  {
+    std::ifstream fs (ff.c_str (), std::ios::in);
 
-  int line = 1;
-  std::string text;
-
-  while (std::getline (fs, text) && line <= end)
+    if (! fs)
+      os << "dbtype: unable to open '" << ff << "' for reading!\n";
+    else
     {
-      if (line >= start)
-        os << line << "\t" << text << "\n";
+      int line = 1;
+      std::string text;
 
-      line++;
+      while (std::getline (fs, text) && line <= end)
+        {
+          if (line >= start)
+            os << line << "\t" << text << "\n";
+
+          line++;
+        }
     }
+  }
 
   os.flush ();
 }
 
 DEFUN (dbtype, args, ,
        "-*- texinfo -*-\n\
 @deftypefn  {} {} dbtype\n\
 @deftypefnx {} {} dbtype @var{lineno}\n\
